### ISSUE TITLE
[DOC]: A bit of guidance through inference please??!

### ISSUE THREAD

**Issue by israa04** [hpcaitech/ColossalAI#3893]
### 📚 The doc issue

After finishing stage 3 of PPO for the Chat application, the state dictionary got saved as per the implementation.
Now I'm trying to make an inference using the `examples/inference.py` file but I'm facing errors regarding loading the saved `state_dict` into the pretrained model.
Here's one instance of the error:
`size mismatch for transformer.word_embeddings.weight: copying a param with shape torch.Size([250880, 64]) from checkpoint, the shape in current model is torch.Size([250880, 4096]).`
and this type of error happens for pretty much all the layers of the model.

I would really appreciate some guidance here as the documentation only mentions serving the model via the interface, which doesn't support the model I fine-tuned (BLOOM).
I tried to load the HuggingFace model as well as the fine-tuned one and they both caused the mentioned errors.
How can I please correctly run the inference? Is it something with the pretrained model? with the state_dict?

**Comment by jiangmingyan** [comment]
> something

Thanks for your advise， you should use '--pretrain' to pass 'bloom-rlhf' .

**Comment by jiangmingyan** [comment]
Thanks for the suggestion, you should pass "--pretrain"  the same as ppo stage.

**Comment by sarrahbbh** [comment]
Yes I did pass it by default in the same way as in the PPO stage, but same error.

**Comment by jiangmingyan** [comment]
> > Thanks for the suggestion, you should pass "--pretrain" the same as ppo stage.
> 
> Yes I did pass it by default in the same way as in the PPO stage, but same error.

Can I take a look at the script you used in the third stage?

**Comment by sarrahbbh** [comment]
Sure, here it is:

`torchrun --standalone --nproc_per_node=4 train_prompts.py --prompt_dataset /pretrain_dataset.json --strategy colossalai_zero2 --model bloom --rm_model bloom --rm_pretrain model/bloom-sft --rm_path /rm_model/bloom-rm.pt --save_path /rlhf_model/bloom-rlhf --need_optim_ckpt True
`

The other arguments that I haven't explicitly passed are set by default in the code
First I had a suspicion that I didn't specify the ```.pt``` extension for the ```--save_path``` argument because the output file was only 30MB, but then I saw that it's used as a state dictionary so it's supposed to be relatively small in size, no?
Thank you for your help!
